public/app/core/services/backend_srv.ts

```diff
@@ -227,6 +179,76 @@ export class BackendSrv implements BackendService {
     });
   }
 
+  private getChunkedResponseObserver({
+    controller,
+    observer,
+    options,
+    requestId,
+  }: {
+    controller: AbortController;
+    observer: Subscriber<FetchResponse<Uint8Array<ArrayBufferLike> | undefined>>;
+    options: BackendSrvRequest;
+    requestId: string;
+  }): Partial<Observer<Response>> {
+    let done = false;
+    return {
+      next: (response) => {
+        const rsp = {
+          status: response.status,
+          statusText: response.statusText,
+          ok: response.ok,
+          headers: response.headers,
+          url: response.url,
+          type: response.type,
+          redirected: response.redirected,
+          config: options,
+          traceId: response.headers.get(GRAFANA_TRACEID_HEADER) ?? undefined,
+          data: undefined,
+        };
+
+        if (!response.body) {
+          observer.next(rsp);
+          observer.complete();
+          return;
+        }
+
+        const reader = response.body.getReader();
+
+        // Setup onabort callback so that we can cancel the reader properly
+        controller.signal.onabort = () => {
+          reader.cancel(controller.signal.reason);
+          console.log(requestId, 'signal.aborted');
+        };
```

Comment:
Replaces this code

```ts
if (controller.signal.aborted) {
  reader.cancel(controller.signal.reason);
  console.log(requestId, 'signal.aborted');
  return;
}
```
Couldn't reach this code in the unit tests because `controller.signal.aborted` would never be true